### PR TITLE
Portfolio user role with patient-measure bootstrapping

### PR DESCRIPTION
**Pull request by jtferns**
#### Stories
- https://www.pivotaltracker.com/story/show/60860100
- https://www.pivotaltracker.com/story/show/63975020
#### Notes
- Added <code>portfolio</code> user role, modeled after existing <code>admin</code> role
- Added corresponding rake tasks, template helper, and measure bootstrapping
## Default user

deafult
## Portfolio user

_Note: Welcome header was used for debugging, has been removed before the commit_
portfolio

### CODE REVIEW

**Comment by andrequina** [comment]
Pete has a way that new patients can get added to all measures... check with him.

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -26,8 +26,10 @@
       <% @patients.each do |patient| %>
         (function() {
           var patient = new Thorax.Models.Patient(<%= patient.to_json(except: :results).html_safe %>, { parse: true });
-          bonnie.measures.each(function(measure) { 
-            if(_(patient.get('measure_ids')).contains(measure.get('hqmf_set_id'))) {
+          bonnie.measures.each(function(measure) {
+            if (<%= current_user.portfolio %>) {
```
Comment (pkmitre) [inline app/views/layouts/application.html.erb]: Using the is_portfolio? method you added to app/models/user.rb would be a bit clearer here.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -42,6 +44,7 @@
         <% end %>
       });
       bonnie.isAdmin = <%= current_user.admin %>;
+      bonnie.isPortfolio = <%= current_user.portfolio %>;
```
Comment (pkmitre) [inline app/views/layouts/application.html.erb]: Here too for is_portfolio?, and I should have used is_admin? in the line above as well.

File: app/assets/javascripts/views/patient_builder/patient_builder.js.coffee [hunk 3]
```diff
@@ -61,6 +61,9 @@ class Thorax.Views.PatientBuilder extends Thorax.View
       sync: (model) ->
         @patients.add model # make sure that the patient exist in the global patient collection
         @measure?.get('patients').add model # and the measure's patient collection
+        if bonnie.isPortfolio
+          @measures.each (m) ->
+            m.get('patients').add model unless m.get('patients').include(model)
```
Comment (pkmitre) [inline app/assets/javascripts/views/patient_builder/patient_builder.js.coffee]: A collection won't add the same model twice, so

```
@measures.each (m) -> m.get('patients').add model
```

would be fine.

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (jtferns) [inline app/views/layouts/application.html.erb]: Agreed.

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (jtferns) [inline app/views/layouts/application.html.erb]: I think that makes sense, I will change the admin line in my update, too.

**Pull request merged**